### DISCUSSION TITLE
Feature Request: Edit the file system as a text buffer

### DISCUSSION THREAD

**Discussion by mmosk** [zed-industries/zed#21847] [Feature Requests]
Hello!

This feature request proposes implementing functionality similar to [oil.nvim](https://github.com/stevearc/oil.nvim) in Zed. In short, oil.nvim allows you to edit the contents of a folder as if it were a text file. When you save the file, changes are applied directly to the file system. Since Zed is a UI-based app, this functionality could also be available to non-Vim users, enabling features like multi-line editing for renaming files. This approach is very convenient, particularly when dealing with similarly named files, such as `auth.controller.js`, `auth.service.js`, and so on.

The expected behavior would be as follows:
- Pressing `-` would switch to the "oil" view in the editor area. Further presses would navigate to parent folders. Unlike oil.nvim, your current location would be synchronized with the explorer view.
- Clicking on a folder name in the explorer would display the list of files in the editor area.
- To fold or unfold a folder in the explorer, you would click the arrow to the left of the folder name.

The last two items essentially describe how GitHub's file explorer works:

<img width="1507" alt="Screenshot 2024-12-11 at 11 33 32" src="https://github.com/user-attachments/assets/2f3bfce2-a733-4ffc-9231-8f751bd73754">

**Comment by 5brian** [comment]
There is an issue open for it, you can like it: https://github.com/zed-industries/zed/issues/20675
